kube-runtime/src/controller/mod.rs

```diff
@@ -1087,41 +1087,38 @@ where
     ///
-    /// For example, this can be used to watch resources once and use the stream to trigger reconciliation and also keep a cache of those objects.
-    /// That way it's possible to use this up to date cache instead of querying Kubernetes to access those resources
+    /// This can be used to inject reconciliations for specific objects from an external  resource.
     ///
     /// # Example:
     ///
     /// ```no_run
     /// # async {
     /// # use futures::{StreamExt, TryStreamExt};
-    /// # use k8s_openapi::api::core::v1::{ConfigMap, Pod};
+    /// # use k8s_openapi::api::core::v1::{ConfigMap};
     /// # use kube::api::Api;
     /// # use kube::runtime::controller::Action;
     /// # use kube::runtime::reflector::{ObjectRef, Store};
     /// # use kube::runtime::{reflector, watcher, Controller, WatchStreamExt};
     /// # use kube::runtime::watcher::Config;
     /// # use kube::{Client, Error, ResourceExt};
+    /// # use tokio_stream::wrappers::IntervalStream;
     /// # use std::future;
+    /// # use std::time::Duration;
     /// # use std::sync::Arc;
     /// #
     /// # let client: Client = todo!();
-    /// # async fn reconcile(_: Arc<ConfigMap>, _: Arc<Store<Pod>>) -> Result<Action, Error> { Ok(Action::await_change()) }
-    /// # fn error_policy(_: Arc<ConfigMap>, _: &kube::Error, _: Arc<Store<Pod>>) -> Action { Action::await_change() }
+    /// # async fn reconcile(_: Arc<ConfigMap>, _: Arc<()>) -> Result<Action, Error> { Ok(Action::await_change()) }
+    /// # fn error_policy(_: Arc<ConfigMap>, _: &kube::Error, _: Arc<()>) -> Action { Action::await_change() }
     /// #
-    /// // Store can be used in the reconciler instead of querying Kube
-    /// let (pod_store, writer) = reflector::store();
-    /// let pod_stream = watcher(Api::<Pod>::all(client.clone()), Config::default())
-    ///     .default_backoff()
-    ///     .reflect(writer)
-    ///     .applied_objects()
-    ///     // Map to the relevant `ObjectRef<K>` to reconcile
-    ///     .map_ok(|pod| ObjectRef::new(&format!("{}-cm", pod.name_any())).within(&pod.namespace().unwrap()));
-    ///
-    /// Controller::new(Api::<ConfigMap>::all(client), Config::default())
-    ///     .reconcile_on(pod_stream)
-    ///     // The store can be re-used between controllers and even inspected from the reconciler through [Context]
-    ///     .run(reconcile, error_policy, Arc::new(pod_store))
+    /// let ns = "external-configs".to_string();
+    /// let externals = [ObjectRef::new("managed-cm1").within(&ns)];
+    /// let mut next_object = externals.into_iter().cycle();
+    /// let interval = tokio::time::interval(Duration::from_secs(60)); // hit the object every minute
+    /// let external_stream = IntervalStream::new(interval).map(|_| Ok(next_object.next().unwrap()));
+    ///
+    /// Controller::new(Api::<ConfigMap>::namespaced(client, &ns), Config::default())
```

Comment:
Shouldn't this be a different namespace to signal that the other resources are completely external?